processing/src/test/java/org/apache/druid/segment/UnnestStorageAdapterTest.java

```diff
@@ -0,0 +1,287 @@
+
+    Sequence<Cursor> cursorSequence = UNNEST_STORAGE_ADAPTER1.makeCursors(
+        null,
+        UNNEST_STORAGE_ADAPTER1.getInterval(),
+        VirtualColumns.EMPTY,
+        Granularities.ALL,
+        false,
+        null
+    );
+    UnnestStorageAdapter adapter = UNNEST_STORAGE_ADAPTER1;
+    Assert.assertEquals(adapter.getDimensionToUnnest(), columnName);
+    Assert.assertEquals(
+        adapter.getColumnCapabilities(OUTPUT_COLUMN_NAME).isDictionaryEncoded(),
+        ColumnCapabilities.Capable.TRUE
+    );
+    Assert.assertEquals(adapter.getMaxValue(columnName), adapter.getMaxValue(OUTPUT_COLUMN_NAME));
+    Assert.assertEquals(adapter.getMinValue(columnName), adapter.getMinValue(OUTPUT_COLUMN_NAME));
+
+    cursorSequence.accumulate(null, (accumulated, cursor) -> {
+      ColumnSelectorFactory factory = cursor.getColumnSelectorFactory();
+
+      DimensionSelector dimSelector = factory.makeDimensionSelector(DefaultDimensionSpec.of(OUTPUT_COLUMN_NAME));
+
+
+      while (!cursor.isDone()) {
+        Object dimSelectorVal = dimSelector.getObject();
+        if (dimSelectorVal == null) {
+          Assert.assertNull(dimSelectorVal);
+        }
+        cursor.advance();
+      }
+      return null;
+    });
+  }
+
+  @Test
+  public void test_dim_unnest_adapters_methods()
+  {
+    final String columnName = "multi-string1";
+
+    Sequence<Cursor> cursorSequence = UNNEST_STORAGE_ADAPTER1.makeCursors(
+        null,
+        UNNEST_STORAGE_ADAPTER1.getInterval(),
+        VirtualColumns.EMPTY,
+        Granularities.ALL,
+        false,
+        null
+    );
+    UnnestStorageAdapter adapter = UNNEST_STORAGE_ADAPTER1;
+    Assert.assertEquals(adapter.getDimensionToUnnest(), columnName);
+    Assert.assertEquals(
+        adapter.getColumnCapabilities(OUTPUT_COLUMN_NAME).isDictionaryEncoded(),
+        ColumnCapabilities.Capable.TRUE
+    );
+    Assert.assertEquals(adapter.getMaxValue(columnName), adapter.getMaxValue(OUTPUT_COLUMN_NAME));
+    Assert.assertEquals(adapter.getMinValue(columnName), adapter.getMinValue(OUTPUT_COLUMN_NAME));
+
+    cursorSequence.accumulate(null, (accumulated, cursor) -> {
+      ColumnSelectorFactory factory = cursor.getColumnSelectorFactory();
+
+      DimensionSelector dimSelector = factory.makeDimensionSelector(DefaultDimensionSpec.of(OUTPUT_COLUMN_NAME));
+
+
+      while (!cursor.isDone()) {
+        Object dimSelectorVal = dimSelector.getObject();
+        Assert.assertNotNull(dimSelector.getRow());
+        Assert.assertNotNull(dimSelector.getValueCardinality());
+        Assert.assertNotNull(dimSelector.makeValueMatcher(OUTPUT_COLUMN_NAME));
+        Assert.assertNotNull(dimSelector.idLookup());
+        Assert.assertNotNull(dimSelector.lookupName(0));
+        Assert.assertNotNull(dimSelector.defaultGetObject());
+        Assert.assertFalse(dimSelector.isNull());
+        if (dimSelectorVal == null) {
+          Assert.assertNull(dimSelectorVal);
+        }
```

Comment:
The assertions here should be updated.  You should be able to know and validate the sepcific ValueCardinality and this seems to always be looking up the value for the 0 index, which shouldn't be correct.  If the test isn't actually walking through rows with different dictionary values, it's not really validating what we need it to.